### ISSUE TITLE
w3lib/encoding.py:45: DeprecationWarning: Flags not at the start of the expression

### ISSUE THREAD

**Issue by mehaase** [scrapy/w3lib#116]
I'm seeing this warning on Python 3.7. Importing the module is enough to show these warnings.

```python
> python -Wall
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site.py:165: DeprecationWarning: 'U' mode is deprecated
  f = open(fullname, "rU")
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site.py:165: DeprecationWarning: 'U' mode is deprecated
  f = open(fullname, "rU")
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site.py:165: DeprecationWarning: 'U' mode is deprecated
  f = open(fullname, "rU")
Python 3.7.0 (default, Sep 12 2018, 18:30:08) 
[GCC 8.0.1 20180414 (experimental) [trunk revision 259383]] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import w3lib.encoding
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site-packages/w3lib/encoding.py:45: DeprecationWarning: Flags not at the start of the expression '<\\s*(?:meta(?x)(?:\\s' (truncated)
  _BODY_ENCODING_STR_RE = re.compile(_BODY_ENCODING_PATTERN, re.I)
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site-packages/w3lib/encoding.py:45: DeprecationWarning: Flags not at the start of the expression '<\\s*(?:meta(?x)(?:\\s' (truncated)
  _BODY_ENCODING_STR_RE = re.compile(_BODY_ENCODING_PATTERN, re.I)
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site-packages/w3lib/encoding.py:46: DeprecationWarning: Flags not at the start of the expression b'<\\s*(?:meta(?x)(?:\\s' (truncated)
  _BODY_ENCODING_BYTES_RE = re.compile(_BODY_ENCODING_PATTERN.encode('ascii'), re.I)
/home/mhaase/.local/share/virtualenvs/starbelly-21k54S-f/lib/python3.7/site-packages/w3lib/encoding.py:46: DeprecationWarning: Flags not at the start of the expression b'<\\s*(?:meta(?x)(?:\\s' (truncated)
  _BODY_ENCODING_BYTES_RE = re.compile(_BODY_ENCODING_PATTERN.encode('ascii'), re.I)
```

This deprecation appears to result from [BPO #22493](https://bugs.python.org/issue22493): inline flags may not be used in the middle of a regex anymore. I cannot see where `encoding.py` is using inline flags, though.

**Comment by mehaase** [comment]
I guess this is the inline flag: `meta(?x)`. But I can't see where this is coming from. If I copy and paste the relevant code into a new file, then the `DeprecationWarning` goes awa.

**Comment by immerrr** [comment]
@exic your deprecation warnings are different, they are about importing abstract base classes (ABCs) that previously lived in `collections` and now reside in `collections.abc` and are scheduled to be removed from their old place in Python 3.8.

Also, curiously, please note that none of those warnings you have shown relates to `w3lib`: two are from `scrapy` and one is from `lxml`.

**Comment by Gallaecio** [comment]
Fixed in e034ba5f30f8e76671ad6ce577a6bc2aff9451f8
